src/library.cpp

```diff
@@ -1489,6 +1489,232 @@ void lammps_scatter_atoms_subset(void *ptr, char *name,
 }
 #endif
 
+
+#if defined(LAMMPS_BIGBIG)
+void lammps_gather_peratom_fix(void *ptr, char * /*id */, int /*type*/, int /*count*/, void * /*data*/)
```

Comment:
i think these functions names would be ok without `peratom`. the `gather` implies that it is about a per-atom property.
you should also add a test whether the fix has the `peratom_flag` and then `size_peratom_cols` will give you the information whether it is a per-atom vector or array. there is also the peratom_freq flag, that provides information when the data is available and thus the function should fail if not.

Reply:
I have now replaced all calls to lmp->atom->extract in `library.cpp` with `lammps_extract_atom` and added lines in this function to check for `property/atom` vectors. This felt like the lightest fix...